libs/positions/src/lib/use-market-position-open-volume.tsx

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useMemo, useState } from 'react';
+import { useVegaWallet } from '@vegaprotocol/wallet';
+import { positionsDataProvider } from './positions-data-providers';
+import { useDataProvider } from '@vegaprotocol/react-helpers';
+import type {
+  PositionsQuery,
+  PositionsSubscriptionSubscription,
+} from './__generated__/Positions';
+
+const getMarketPosition = ({
+  data,
+  marketId,
+}: {
+  data: PositionsQuery['party'];
+  marketId: string;
+}) => {
+  const positions =
+    data?.positionsConnection?.edges?.map((item) => item.node) ?? [];
+  return positions.find((item) => item.market.id === marketId);
+};
+
+export const useMarketPositionOpenVolume = (marketId: string) => {
+  const { pubKey } = useVegaWallet();
+  const [openVolume, setOpenVolume] = useState<string>('');
+  const variables = useMemo(() => {
+    return { partyId: pubKey || '' };
+  }, [pubKey]);
+  const update = useCallback(
```

Comment:
yeah, actually it could be handled also like you said, because it returns just single string, but in that way like it is here it's consistent with other similar hooks, which return more complex object and can be handled only by using _useState_'s for better memoization reason.